pkg/clients/database/mysql.go

```diff
@@ -338,7 +338,7 @@ func IsMySQLUpToDate(p azuredbv1beta1.SQLServerParameters, in mysql.Server) bool
 		return false
 	}
 	switch {
-	case p.MinimalTLSVersion != string(in.MinimalTLSVersion):
+	case p.MinimalTLSVersion != string(in.MinimalTLSVersion) && p.SSLEnforcement != string(mysql.SslEnforcementEnumDisabled):
```

Comment:
@sergenyalcin do we have a validation on `SSLEnforcement` that forces the enum constants defined in the SDK, e.g., `mysql.SslEnforcementEnumDisabled`?

Reply:
Yes we have a validation for this field:

https://github.com/crossplane/provider-azure/blob/3bbc32a92e59cd0a57eeefc1acaf24fc59f51221/apis/database/v1beta1/sql_types.go#L194